apps/web/src/contexts/bottomsheet/BottomSheetContext.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { createContext, memo, useCallback, useContext, useMemo, useState } from 'react';
+import colors from 'shared/theme/colors';
+import styled from 'styled-components';
+import { Drawer } from 'vaul';
+
+type BottomSheetState = {
+  isBottomSheetModalVisible: boolean;
+};
+const BottomSheetStateContext = createContext<BottomSheetState | undefined>(undefined);
+
+export const useBottomSheetModalState = (): BottomSheetState => {
+  const context = useContext(BottomSheetStateContext);
+  if (!context) {
+    throw new Error('Attempted to use BottomSheetModalStateContext without a provider!');
+  }
+  return context;
+};
+
+type BottomSheetActions = {
+  showBottomSheet: (bottomSheetModal: BottomSheet) => void;
+  hideBottomSheet: () => void;
+};
+
+const BottomSheetActionsContext = createContext<BottomSheetActions | undefined>(undefined);
+
+export const useBottomSheetActions = (): BottomSheetActions => {
+  const context = useContext(BottomSheetActionsContext);
+  if (!context) {
+    throw new Error('Attempted to use BottomSheetModalActionsContext without a provider!');
+  }
+  return context;
+};
+
+type BottomSheetModalProviderProps = {
+  children: React.ReactNode;
+};
```

Comment:
hypernit: for situations like this, we can actually just use `PropsWithChildren` and import directly from react: `import { PropsWithChildren } from 'react';`

probably a few places to refactor in our repo, no need to handle in this PR :)

Reply:
oh maybe this will mess with the `BottomSheet` type you defined below, ignore if so

Reply:
Seems like it's messed up with the types 🤦🏻‍♂️